src/assets/content.js

```diff
@@ -5,13 +5,56 @@
 // from the contents of the active page
 chrome.runtime.onMessage.addListener( function(request, sender, sendResponse) {
   if (request.message === 'send links') {
-    sendResponse(Array.from(document.links).map( function(val) {
-      return {
-        'href': val.href,
-        'host': val.host,
-        'tagName': val.tagName,
-        'hidden': val.hidden
-      };
-    }));
+    sendResponse(findLinks());
   }
 });
+
+function findLinks() {
+  const urlRegex = /https?:\/\/[^\s]+/;
+  const all = document.querySelectorAll('*');
+  const links = [];
+  all.forEach((element) => {
+    if (element.getAttribute('href')) {
+      if (element.tagName == 'A' || element.tagName == 'AREA') {
+        links.push(getLinkData(element, element.href));
+      }
+    } else if (element.hasAttribute('onclick')) {
+      const jsFunc = element.getAttribute('onclick');
+      const specialCase = /window[.]location[.]href=["'](.*)["']/;
+      const specialCaseMatches = jsFunc.match(specialCase);
+      const generalMatches = jsFunc.match(urlRegex);
+      if (specialCaseMatches) {
+        let urlString = specialCaseMatches[1];
+        if (!urlRegex.test(urlString)) {
+          urlString = window.location.origin + urlString;
+        }
+        links.push(getLinkData(element, urlString));
+      } else if (generalMatches) {
+        links.push(getLinkData(element, generalMatches[0]));
+      }
+    } else if (element.hasAttribute('action')) {
```

Comment:
Does action can always be associated with a form element?

Reply:
Yes, I could add a comment though to clarify that

Reply:
Although if you mean looping through the elements three time then we would lose the order of the DOM elements